docs/proposals/sysctl.md

```diff
@@ -0,0 +1,740 @@
+
+**Note**: with the exception of `kernel.shm*` all of the listed pod-level sysctls depend on kernel memory accounting to be enabled for proper resource isolation. This will not be the case for 1.4 by default, but is planned in 1.5.
+
+**Note**: all the ipc objects persist when the originating containers dies. Their resources (if kmem accounting is enabled) fall back to the parent cgroup. As long as there is no pod level memory cgroup, the parent will be the container runtime, e.g. the docker daemon or the RunC process. It is [planned with v1.5 to introduce a pod level memory cgroup](pod-resource-management.md#implementation-status) which will fix this problem.
+
+**Note**: in general it is good practice to reserve special nodes for those pods which set sysctls which the kernel does not guarantee proper isolation for.
+
+## Proposed Design
+
+Sysctls in pods and `PodSecurityPolicy` are first introduced as an alpha feature for Kubernetes 1.4. This means that the API will model these as annotations, with the plan to turn those in first class citizens in a later release when the feature is promoted to beta.
+
+It is proposed to use a syntactical validation in the apiserver **and** a node-level whitelist of *safe sysctls* in the kubelet. The whitelist shall be fixed per version and might grow in the future when better resource isolation is in place in the kubelet. In addition a list of *allowed unsafe sysctls* will be configured per node by the cluster admin, with an empty list as the default.
+
+The following rules apply:
+
+- Only sysctls shall be whitelisted in the kubelet
+  + that are properly namespaced by the container or the pod (e.g. in the ipc or net namespace)
+  + **and** that cannot lead to resource consumption outside of the limits of the container or the pod.
+  These are called *safe*.
+- The cluster admin shall only be able to manually enable sysctls in the kubelet
+  + that are properly namespaced by the container or the pod (e.g. in the ipc or net namespace).
+  These are call *unsafe*.
+
+This means that sysctls that are not namespaced must be set by the admin on host level at his own risk, e.g. by running a *privileged daemonset*, possibly limited to a restricted, special-purpose set of nodes, if necessary with the host network namespace. This is considered out-of-scope of this proposal and out-of-scope of what the kubelet will do for the admin. A section is going to be added to the documentation describing this.
+
+The *allowed unsafe sysctls* will be configurable on the node via a flag of the kubelet.
+
+### Pod API Changes
+
+Pod specification must be changed to allow the specification of kernel parameters:
+
+```go
+// Sysctl defines a kernel parameter to be set
+type Sysctl struct {
+	// Name of a property to set
+	Name string `json:"name"`
+	// Value of a property to set
+	Value string `json:"value"`
```

Comment:
That's probably more a UX issue:

```yaml
sysctl:
  name: kernel.shmmax
  value: 1234567
```

instead of

```yaml
    value: "1234567"
```

So I agree, should be IntOrString.